version: '3.3'
services:
  db:
    image: postgres:14.1-alpine
    command: [
      "postgres", 
      "-c", 
      "config_file=/etc/postgresql.conf",
      "-c",
      "hba_file=/etc/pg_hba.conf"
    ]
    restart: always
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_DB=test_db
    ports:
    # почему-то на моем компьютере не работает порт 5432
      - '54321:5432'
    volumes: 
      - db_volume:/var/lib/postgresql/data
      - ./database/pg_hba.conf:/etc/pg_hba.conf
      - ./database/postgresql.conf:/etc/postgresql.conf
  server:
    build: ./django/
    command: python manage.py runserver 0.0.0.0:8001
    volumes:
      - ./django:/code
    ports:
      - "8001:8001"
      - "8000:8000"
    environment:
      - POSTGRES_NAME=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_PORT=5432
      - DJANGO_ALLOWED_HOSTS=localhost 127.0.0.1 195.14.48.196 192.168.120.25
      - DEBUG=1
    depends_on:
      - db
  frontend:
    build: ./frontend/
    volumes:
      - ./frontend:/app
    ports:
      - "80:80"
    depends_on:
      - server
      
volumes:
  db_volume: